packages/worker/src/worker.ts

```diff
@@ -924,9 +928,17 @@ export class Worker {
         ),
         { defaultValue: undefined }
       );
-    } finally {
+      // Only shutdown the native worker if we completed without an error.
+      // Otherwise Rust / TS are in an unknown state and shutdown might hang.
+      // A new process must be created in order to instantiate a new Rust Core.
+      // TODO: force shutdown in core?
       await this.nativeWorker.completeShutdown();
-      await this.workflowCreator?.destroy();
+    } finally {
+      try {
+        await this.workflowCreator?.destroy();
```

Comment:
Any concerns of, in an erroring situation, this throws and swallows the original error (assuming that's what JS does, I forget)? I can understand that an error here is very fatal, but unwinding from what may be a very fatal error in the original try, I wonder if this will hide what the real fatal error was.

Reply:
Luckily we log the error before we throw it